### ISSUE TITLE
Error on certificate validation on the last few days

### ISSUE THREAD

**Issue by SandroMachado** [babylonhealth/certificate-transparency-android#34]
My certificate is about to expire and and on the last few days, some instances of my application are having problems connecting to the server. They just don't connect. Even on the same device, sometimes after installing the app it works fine, other times it just doesn't connect.

Digging in I discovered that removing this library fixes the issue. Any  tip on how to fix this? 

Or any ideas why this is happening?

**Comment by sheungon** [comment]
Same issue happened to me few days ago. I had to disable CT to get my app work again.

**Comment by pavlospt** [comment]
@mattmook are there any news regarding this? We started noticing similar behaviour today in our app :)

**Comment by SandroMachado** [comment]
Same here.

**Comment by jclaudino** [comment]
Same

**Comment by barnhill** [comment]
Noticing same here

**Comment by maxkohne** [comment]
We are experiencing this as well

**Comment by eramirez01** [comment]
Same Here!!!

**Comment by pavlospt** [comment]
Okay this sounds like something is certainly going on. Just to add more info here, our iOS colleagues were able to use CT fine the whole day, while we had those issues! So i guess it is not infrastructure related.

**Comment by mattmook** [comment]
Has anyone managed to capture a log from when this happens? I suspect it could be something like the log list being out of sync with the signature, which we have seen happen once before.

**Comment by pavlospt** [comment]
Hey @mattmook , I do not happen to have a log, but the behavior of failing to verify the signatures was like consistently inconsistent. At some it would fail to verify and then would work again. I am not sure if it has something to do with the last days of the month and an issue is going on with Date validations, but we have not seen it happen afterwards. 

Just for future reference and in order to help you debug it more effectively, what would you like from us to capture? Just the raw loglist response?

**Comment by mattmook** [comment]
When creating the `certificateTransparencyInterceptor` you can provide a logger, an instance of [`CTLogger`](https://github.com/babylonhealth/certificate-transparency-android/blob/master/lib/src/main/kotlin/com/babylon/certificatetransparency/CTLogger.kt) - it might be worth recording any instances when `log` is called with an instance of `VerificationResult.Failure`.

**Comment by pavlospt** [comment]
Yeap, we are already using that and it was outputting something along the lines of: 
`SSLHandshakeException: Certificate transparency failed: LogServerSignatureResult.Invalid.SignatureFailed` I can get the exact log next time it happens 😄

**Comment by blinkmacalahan** [comment]
I'm not sure if related or not, but I recently added this library and things were working fine. Starting today, cert transparency is failing:

`Failure: Unable to load log servers with SignatureVerificationFailed(signatureResult=Invalid signature)`

**Comment by jclaudino** [comment]
Any updates on this issue? 

Are others still experiencing this?

**Comment by pavlospt** [comment]
Haven't experienced it since the last days of January. Will wait to check it on February as well!

**Comment by pavlospt** [comment]
We had this logged in production today as well. Unfortunately we do not have any logs to share apart from:
```
javax.net.ssl.SSLPeerUnverifiedException: Certificate transparency failed at com.babylon.certificatetransparency.internal.verifier.CertificateTransparencyInterceptor.intercept(CertificateTransparencyInterceptor.kt:59)
```

**Comment by maxkohne** [comment]
Yeah, we actually disabled this library since we cannot afford to have these issues happen randomly. Our security team understands but would really like this to be solved / looked into.

I wrote about how we are remotely toggled this library in another issue: https://github.com/babylonhealth/certificate-transparency-android/issues/28#issuecomment-582993663

**Comment by pavlospt** [comment]
@mattmook based on this comment: https://github.com/babylonhealth/certificate-transparency-android/issues/28#issuecomment-582653499 could we please have an update that uses the correct casing for the Signature name `SHA256withRSA` https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#Signature and prepare a new release? 

Also since more people mentioned issues with the last days of a month, I really tend to believe something shady is going on there.

**Comment by vivascu** [comment]
There are some interesting findings I mentioned here: https://github.com/babylonhealth/certificate-transparency-android/issues/28#issuecomment-625865487
